src/stores/orderListStore.ts

```diff
@@ -54,15 +55,37 @@ class OrderListStore extends BaseStore {
   }
 
   fetchAll = async () => {
+    this.hasPendingOrders = true;
     const dto = await this.api.fetchAll();
+    this.orders = dto.map(mappers.mapToLimitOrder);
+    this.hasPendingOrders = false;
+  };
 
-    this.updateOrders(dto.map(mappers.mapToLimitOrder));
-    this.rootStore.orderBookStore.fetchAll();
+  @action
+  addOrder = (order: any) => {
+    this.orders.push(mappers.mapToLimitOrder(order));
   };
 
   @action
-  updateOrders = (orders: OrderModel[]) => {
-    this.orders = orders;
+  deleteOrder = (orderId: string) => {
+    this.orders.some((existedOrder: OrderModel, index: number) => {
+      if (existedOrder.id === orderId) {
+        this.orders.splice(index, 1);
+        return true;
+      }
+      return false;
+    });
+  };
+
+  @action
+  updateOrder = (dto: any) => {
```

Comment:
actually, this is `addOrUpdate` or `merge` function, so it would be nice to rename or just to do only one thing